sus/ptr/copy.h

```diff
@@ -79,9 +79,12 @@ void copy_nonoverlapping(::sus::marker::UnsafeFnMarker, const T* src, T* dst,
                   (dst < src && dst <= src - count.primitive_value));
   if constexpr (::sus::mem::size_of<T>() > 1) {
     auto bytes = count.checked_mul(::sus::mem::size_of<T>()).expect("overflow");
-    memcpy(dst, src, bytes);
+    // Clang isn't able to detect that `T` is trivially relocatable, and errors
+    // out when it's not trivially copyable. We cast to `void*` since we are
+    // certain that it's trivially relocatable.
```

Comment:
```suggestion
    // out when it's not trivially copyable. We cast to `void*` since we
    // require the caller to ensure the type is `TrivialCopy` or
    // `TriviallyRelocatable`.
```

Looks like we should do the same in `copy()` even though it doesn't happen to trip up clang in our tests?